File: GXpert/GXpert.Web/Modules/Masters/UserSkillsScore/UserSkillsScoreColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace GXpert.Masters.Columns;

[ColumnsScript("Masters.UserSkillsScore")]
[BasedOnRow(typeof(UserSkillsScoreRow), CheckNames = true)]
public class UserSkillsScoreColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string SkillCategoryName { get; set; }
    public string SkillName { get; set; }
    public float Score { get; set; }   
}

File: GXpert/GXpert.Web/Modules/ServerTypes/Masters/UserSkillsScoreColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { UserSkillsScoreRow } from "./UserSkillsScoreRow";

export interface UserSkillsScoreColumns {
    Id: Column<UserSkillsScoreRow>;
    SkillCategoryName: Column<UserSkillsScoreRow>;
    SkillName: Column<UserSkillsScoreRow>;
    Score: Column<UserSkillsScoreRow>;
}

export class UserSkillsScoreColumns extends ColumnsBase<UserSkillsScoreRow> {
    static readonly columnsKey = 'Masters.UserSkillsScore';
    static readonly Fields = fieldsProxy<UserSkillsScoreColumns>();
}